### PR TITLE
defined User and Address models

### PR DESCRIPTION
**Pull request by mh-393**
Defined User and Address models including validation hooks for phone number and zip code.

### CODE REVIEW

File: server/db/models/user.js [hunk 1]
```diff
@@ -3,10 +3,53 @@ const Sequelize = require('sequelize')
...
-    allowNull: false
+    allowNull: false,
+    validate: {
+      notEmpty: true,
+      isEmail: true
+    }
+  },
+  phone: {
+    type: Sequelize.INTEGER,
+    unique: true,
```
Comment (kennethchen91) [inline server/db/models/user.js]: Shouldn't be unique -- different people in same household can have same landline.

File: server/db/models/user.js [hunk 2]
```diff
@@ -3,10 +3,53 @@ const Sequelize = require('sequelize')
...
+    allowNull: false,
+    validate: {
+      notEmpty: true,
+      isNumeric: true,
+      max: 10,
+      min: 10
+    }
+  },
+  defaultShipping: {
```
Comment (kennethchen91) [inline server/db/models/user.js]: We should consider if we want to allow null and empty on this.
What if address instance doesn't exist at time of creating user?

File: server/db/models/user.js [hunk 3]
```diff
@@ -3,10 +3,53 @@ const Sequelize = require('sequelize')
...
+    type: Sequelize.STRING,
+    allowNull: false,
+    validate: {
+      notEmpty: true
+    }
+  },
+  defaultBilling: {
```
Comment (kennethchen91) [inline server/db/models/user.js]: We should consider if we want to allow null and empty on this.
What if address instance doesn't exist at time of creating user?

File: server/db/models/user.js [hunk 4]
```diff
@@ -63,6 +106,11 @@ const setSaltAndPassword = user => {
   }
 }
 
+const validatePhone = user => {
+  user.phone = +user.phone.replace(/[-]/g, '')
```
Comment (kennethchen91) [inline server/db/models/user.js]: Just add replacements for characters like parentheses, dots, and spaces. 
Or just make the whole regex swap out any non-numeric?

**Comment by adamvare** [comment]
Items resolved.

**Pull request merged**